app/webpacker/components/Panel/pages/MergeUsersPage/index.jsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState } from 'react';
+import { Button, Header, Modal } from 'semantic-ui-react';
+import AdminWcaSearch from '../../../SearchWidget/AdminWcaSearch';
+import SEARCH_MODELS from '../../../SearchWidget/SearchModel';
+import useInputState from '../../../../lib/hooks/useInputState';
+import MergeUsers from './MergeUsers';
+
+export default function MergeUsersPage() {
+  const [firstUserId, setFirstUserId] = useInputState(6387);
+  const [secondUserId, setSecondUserId] = useInputState(6388);
+  const [modalOpen, setModalOpen] = useState(false);
+
+  return (
+    <>
+      <Header>Merge Users</Header>
+      <Header as="h4">Select User 1</Header>
+      <AdminWcaSearch
+        label="Search User"
+        model={SEARCH_MODELS.user}
+        multiple={false}
+        value={firstUserId}
+        onChange={setFirstUserId}
+      />
+      <Header as="h4">Select User 2</Header>
+      <AdminWcaSearch
+        label="Search User"
+        model={SEARCH_MODELS.user}
+        multiple={false}
+        value={secondUserId}
+        onChange={setSecondUserId}
+      />
+      <Button
+        primary
+        disabled={!firstUserId || !secondUserId || firstUserId === secondUserId}
+        onClick={() => setModalOpen(true)}
+      >
+        Initiate Merge
+      </Button>
+      <Modal open={modalOpen}>
+        <Modal.Content>
+          <MergeUsers firstUserId={firstUserId} secondUserId={secondUserId} />
```

Comment:
It feels a bit redundant to pass in the IDs and then fetch the user details again separately inside the `MergeUsers` component. The `AdminWcaSearch` component unfortunately uses `IdWcaSearch` under the hood, but you could modify it so that it takes an additional `idOnly` boolean flag as param, and when you set that to `false` it can patch through the whole model. Doing so would save you two network calls.

Reply:
The component requires few other data as well like competitions_organized, competitions_delegated, roles, etc. So if I make AdminSearch to fetch these, it will do these calls in other places as well.

Also, I'm planning to reuse this component in DuplicatePersons page. If the user clicks 'merge' against one of the duplicate, I'm planning to show this view in a modal. At that time as well, I might not have all the necessary data.

So is it fine to go with the current way?

Reply:
Okay, in case of extra data and the potential for reuse later, then it's fine